src/components/map-component.tsx

```diff
@@ -169,44 +190,73 @@ export class MapComponent extends BaseComponent<IProps, IState>{
-            <Popup>
-              Popup for any custom information.
-            </Popup>
-          </Marker>
-          {cityItems}
-          <TileLayer
-              url="https://maps.wikimedia.org/osm-intl/{z}/{x}/{y}@2x.png"
-          />
-          <MapTephraThicknessLayer
-            ref={this.tephraRef}
-            corner1Bound={corner1}
-            corner2Bound={corner2}
-            viewportBounds={viewportBounds}
-            volcanoPos={volcanoPos}
-            gridSize={1}
-            map={mapRef}
-            windSpeed={windSpeed}
-            windDirection={windDirection}
-            colHeight={colHeight}
-            mass={mass}
-            particleSize={particleSize}
+          <ScaleControl
+            position="topright"
           />
-          { showCrossSectionSelector && <CrossSectionDrawLayer
-            ref={this.crossRef}
-            map={mapRef}
-            p1Lat={crossPoint1Lat}
-            p2Lat={crossPoint2Lat}
-            p1Lng={crossPoint1Lng}
-            p2Lng={crossPoint2Lng}
-          /> }
+          <Pane
+            style={{zIndex: 0}}>
+            <TileLayer
+                url="https://maps.wikimedia.org/osm-intl/{z}/{x}/{y}@2x.png"
+            />
+            <MapTephraThicknessLayer
+              ref={this.tephraRef}
+              corner1Bound={corner1}
+              corner2Bound={corner2}
+              viewportBounds={viewportBounds}
+              volcanoPos={volcanoPos}
+              gridSize={1}
+              map={mapRef}
+              windSpeed={windSpeed}
+              windDirection={windDirection}
+              colHeight={colHeight}
+              mass={mass}
+              particleSize={particleSize}
+            />
+          </Pane>
+          <Pane
+            style={{zIndex: 3}}>
+            <Marker
+              position={[volcanoLat, volcanoLng]}
+              icon={iconVolcano}>
+              <Popup>
+                Popup for any custom information.
+              </Popup>
+            </Marker>
+            {cityItems}
+            { showCrossSectionSelector && <CrossSectionDrawLayer
+              ref={this.crossRef}
+              map={mapRef}
+              p1Lat={crossPoint1Lat}
+              p2Lat={crossPoint2Lat}
+              p1Lng={crossPoint1Lng}
+              p2Lng={crossPoint2Lng}
+            /> }
+            {showRuler && <RulerDrawLayer
+              map={mapRef}
+            />}
+          </Pane>
         </LeafletMap>
+        <OverlayControls
+          showRuler={showRuler}
+          onRulerClick={this.onRulerClick}
+          isSelectingCrossSection={isSelectingCrossSection}
+          showCrossSection={showCrossSection}
+          onCrossSectionClick={this.onCrossSectionClick}
+        />
       </CanvDiv>
     );
   }
 
+  private onRulerClick = () => {
+    this.setState({showRuler: !this.state.showRuler});
+    this.stores.setCrossSectionSelectorVisibility(false);
+    this.stores.setIsSelectingCrossSection(false);
```

Comment:
`onRulerClick` changes the local state, and then invokes two actions on the stores.  

Because the cross-section view and ruler views need to know about it each other, it may be easier to do this with explicit actions & props in the stores.